sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -125,6 +125,14 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
             } else if (clientException != null &&
                 WebExceptionUtility.isReadTimeoutException(clientException) &&
                 Exceptions.isSubStatusCode(clientException, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT)) {
+
+                boolean canFailoverOnTimeout = gatewayRequestCanFailoverOnTimeout(request);
+
+                //if operation is data plane, metadata read, or query plan it can be retried on a different endpoint.
+                if(canFailoverOnTimeout) {
+                    return shouldRetryOnEndpointFailureAsync(this.isReadRequest, true, true);
+                }
+
                 // if operationType is QueryPlan / AddressRefresh then just retry
                 if (this.request.getOperationType() == OperationType.QueryPlan || this.request.isAddressRefresh()) {
```

Comment:
queryPlan is being handled above, should we remove the queryPlan handling logic here?

Reply:
I think we can